### PR TITLE
Predicates for Service, Bundle, and Framework Events

### PR DESCRIPTION
**Pull request by stbischof**
Utilities to Filter Service Bundle and FrameworkEvents. Usefull for further Asserations and in combination with EventRecording.
If this fits to `osgi-test` i will add some docs, otherwise we can close this.

### CODE REVIEW

**Comment by bjhargrave** [comment]
Can you rebase on main and also fix the copyright comments for the new format in this Eclipse project?

**Comment by stbischof** [comment]
rebase on main and also switched the copyright header to Eclipse.

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+public interface EventPredicates {
+
+	static <E> Predicate<E> any() {
+		return element -> true;
+	}
+
+	static <E> Predicate<Optional<E>> isPresentAnd(final Predicate<E> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:46]: For the all the method `Predicate` arguments, they should be `Predicate<? super E>`. This allows many more valid predicates to be used.

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+			.noneMatch(predicate);
+	}
+
+	@SafeVarargs
+	static <E> Predicate<List<E>> ordered(final Predicate<E> predicate, final Predicate<E>... predicatesNext) {
+		return elements -> {
+
+			List<Predicate<E>> predicates = new ArrayList<>();
+			predicates.add(predicate);
+			predicates.addAll(Arrays.asList(predicatesNext));
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:83]: This should be `Collections.addAll(predicates, predicatesNext);`

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+		};
+	}
+
+	static <E> Predicate<List<E>> hasSize(long size, final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.filter(predicate)
+			.count() == size;
+	}
+
+	static <E> Predicate<List<E>> hasMoreThen(long size, final Predicate<E> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:121]: hasMoreThan

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+		return elements -> elements.stream()
+			.filter(predicate)
+			.count() > size;
+	}
+
+	static <E> Predicate<List<E>> hasLessThen(long size, final Predicate<E> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:127]: hasMoreThan

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+		static Predicate<FrameworkEvent> isTypeStartlevelChanged() {
+			return e -> isType(FrameworkEvent.STARTLEVEL_CHANGED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStopped() {
+			return e -> isType(FrameworkEvent.STOPPED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStoppedBootclasspathModified() {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:173]: Don't include this. It was deprecated in Core R8.

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+				return false;
+			};
+
+		}
+
+		public static Predicate<ServiceEvent> containServiceProperty(final String key, Object value) {
+			return e -> containsServiceProperties(Dictionaries.dictionaryOf(key, value)).test(e);
+		}
+
+		public static Predicate<ServiceEvent> containsServiceProperties(Map<String, Object> map) {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:241]: The map argument should be `Map<String,?>`. In general, for arguments, we should use the most expansive generic types. All other method arguments need similar changes.

**Review by bjhargrave — COMMENTED** [review]
Initial comments.

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 6 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:241]: using `Map<String,?>`

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 5 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:173]: removed

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 4 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:127]: fixed typo

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 3 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:121]: fixed typo

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 2 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:83]: thx

File: org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java (hunk 1 above)
Comment (stbischof) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java:46]: switched to `Predicate<? super E>`

**Comment by stbischof** [comment]
@bjhargrave I fixes you initial comments.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/bundleevent/BundleEventpredicatesTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+package org.osgi.test.assertj.test.bundleevent;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleEvent;
+import org.osgi.test.assertj.bundleevent.BundleEventPredicates;
+
+public class BundleEventpredicatesTest {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/bundleevent/BundleEventpredicatesTest.java:28]: Capitalization error: BundleEventPredicatesTest

File: org.osgi.test.assertj.framework/test.bndrun [hunk 8]
```diff
@@ -50,9 +50,9 @@
 	junit-platform-commons;version='[1.7.2,1.7.3)',\
 	junit-platform-engine;version='[1.7.2,1.7.3)',\
 	junit-platform-launcher;version='[1.7.2,1.7.3)',\
-	net.bytebuddy.byte-buddy;version='[1.11.2,1.11.3)',\
-	net.bytebuddy.byte-buddy-agent;version='[1.11.2,1.11.3)',\
-	org.mockito.mockito-core;version='[3.11.1,3.11.2)',\
+	net.bytebuddy.byte-buddy;version='[1.11.3,1.11.4)',\
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/test.bndrun:53]: This should not be part of this PR.

**Review by bjhargrave — COMMENTED** [review]
I think some more test are necessary.

File: org.osgi.test.assertj.framework/test.bndrun (hunk 8 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/test.bndrun:53]: sure.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/bundleevent/BundleEventpredicatesTest.java (hunk 7 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/bundleevent/BundleEventpredicatesTest.java:28]: renamed

**Comment by stbischof** [comment]
I added a more tests.

**Comment by stbischof** [comment]
@bjhargrave This PR Seems complete to me. Could you please review?

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	/**
+	 * Returns a predicate that tests if the object Bundle event and matches the
+	 * given eventTypeMask.
+	 *
+	 * @param predicate the predicate
+	 * @return the predicate
+	 */
+	public static Predicate<Object> bundleEventAnd(Predicate<BundleEvent> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java:50]: Why wouldn't someone just use the existing Predicate.and method? This seems to duplicate that existing method.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+		return e -> (e.getType() & eventTypeMask) != 0;
+	}
+
+	/**
+	 * Returns a predicate that tests if the event-type is installed.
+	 *
+	 * @return the predicate
+	 */
+	// BundleEvents - by type
+	public static Predicate<BundleEvent> typeInstalled() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java:71]: I don't like the explosion of method names with event types in them.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	/**
+	 * Returns a predicate that tests if the object is a framework event and the
+	 * given predicate also matches.
+	 *
+	 * @param predicate the predicate
+	 * @return the predicate
+	 */
+	public static Predicate<Object> frameworkEventAnd(Predicate<FrameworkEvent> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java:47]: See and comment above

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+	public static Predicate<FrameworkEvent> type(final int eventTypeMask) {
+		return e -> (e.getType() & eventTypeMask) != 0;
+	}
+
+	/**
+	 * Returns a predicate that tests if the event-type is error.
+	 *
+	 * @return the predicate
+	 */
+	public static Predicate<FrameworkEvent> typeError() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java:67]: See comment about event names in method names.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+		return e -> e instanceof ServiceEvent;
+	}
+
+	/**
+	 * Returns a predicate that tests if the object matches the given predicate.
+	 *
+	 * @param predicate the predicate
+	 * @return the predicate
+	 */
+	public static Predicate<Object> serviceEventAnd(Predicate<ServiceEvent> predicate) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:56]: and comment

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+	public static Predicate<ServiceEvent> type(final int eventTypeMask) {
+		return e -> (e.getType() & eventTypeMask) != 0;
+	}
+
+	/**
+	 * Returns a predicate that tests if the event-type is registered.
+	 *
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> typeRegistered() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:77]: method name comment

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+	private BundleEventPredicates() {}
+
+	// BundleEvents
+
+	/**
+	 * Returns a predicate that tests if the object Bundle event.
+	 *
+	 * @return the predicate
+	 */
+	public static Predicate<Object> bundleEvent() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java:39]: I think we may want to declare this as `public static <T> Predicate<T> bundleEvent()`. This would then allow someone to chain predicates and not lose the type.

```java
Predicate<BundleEvent> p = bundleEvent().and(type(BundleEvent.INSTALLED));
```

But I am not sure how useful this would be in practice.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+package org.osgi.test.assertj.frameworkevent;
+
+import java.util.function.Predicate;
+
+import org.osgi.framework.FrameworkEvent;
+
+/**
+ * The Class FrameworkEventPredicates.
+ */
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java:27]: `@since`

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * The Class ServiceEventPredicates.
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:36]: `@since`

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+	 * @param objectClass the objectClass
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> objectClass(final Class<?> objectClass) {
+
+		return e -> {
+			Object classes = e.getServiceReference()
+				.getProperty(Constants.OBJECTCLASS);
+
+			if (classes != null && classes instanceof String[]) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:76]: You don't need a null check since null is never an instanceof anything. Even the instanceof check is sort of superfluous unless you are defending against rouge ServiceEvent/ServiceReference implementations.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+	/**
+	 * Returns a predicate that tests if the object Contain the service
+	 * property.
+	 *
+	 * @param key the key
+	 * @param value the value
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> containServiceProperty(final String key, Object value) {
+		return e -> containsServiceProperties(Dictionaries.dictionaryOf(key, value)).test(e);
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:95]: You can just 

```java
return containsServiceProperties(Dictionaries.dictionaryOf(key, value));
```

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+	 * Returns a predicate that tests if the object Contains the service
+	 * properties.
+	 *
+	 * @param map the map
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> containsServiceProperties(Map<String, ?> map) {
+		return e -> {
+			ServiceReference<?> sr = e.getServiceReference();
+			List<String> keys = Stream.of(sr.getPropertyKeys())
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:108]: Isn't this just `Arrays.asList(sr.getPropertyKeys())`?

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+
+	/**
+	 * Returns a predicate that tests if the object Contains the service
+	 * properties.
+	 *
+	 * @param properties the properties
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> containsServiceProperties(Dictionary<String, ?> properties) {
+		return e -> containsServiceProperties(Dictionaries.asMap(properties)).test(e);
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:130]: You can just 

```java
return containsServiceProperties(Dictionaries.asMap(properties));
```

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 22]
```diff
@@ -0,0 +1,173 @@
...
+	/**
+	 * Returns a predicate that tests if the object Matches matches the given
+	 * parameters.
+	 *
+	 * @param eventTypeMask the event type mask
+	 * @param objectClass the objectClass
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> matches(int eventTypeMask, final Class<?> objectClass) {
+		return e -> type(eventTypeMask).test(e) && (objectClass(objectClass).test(e));
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:142]: Isn't this just

```java
return type(eventTypeMask).and(objectClass(objectClass));
```

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 23]
```diff
@@ -0,0 +1,173 @@
...
+	 *
+	 * @param eventTypeMask the event type mask
+	 * @param objectClass the objectClass
+	 * @param properties the properties
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> matches(int eventTypeMask, final Class<?> objectClass,
+		Map<String, ?> properties) {
+		return e -> type(eventTypeMask).test(e) && (objectClass(objectClass).test(e))
+			&& containsServiceProperties(properties).test(e);
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:157]: Isn't this just

```java
return type(eventTypeMask).and(objectClass(objectClass)).and(containsServiceProperties(properties));
```

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java [hunk 24]
```diff
@@ -0,0 +1,173 @@
...
+	 * parameters.
+	 *
+	 * @param eventTypeMask the event type mask
+	 * @param objectClass the objectClass
+	 * @param properties the properties
+	 * @return the predicate
+	 */
+	public static Predicate<ServiceEvent> matches(int eventTypeMask, final Class<?> objectClass,
+		Dictionary<String, ?> properties) {
+		return e -> matches(eventTypeMask, objectClass, Dictionaries.asMap(properties)).test(e);
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:171]: Isn't this just

```java
return matches(eventTypeMask, objectClass, Dictionaries.asMap(properties));
```

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+
+package org.osgi.test.assertj.bundleevent;
+
+import java.util.function.Predicate;
+
+import org.osgi.framework.BundleEvent;
+
+/**
+ * The Interface BundleEventPredicates.
+ */
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java:27]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+public class BundleEventPredicates {
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java:28]: fixes for all 3 Predicates

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java (hunk 9 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java]: removed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java (hunk 10 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java]: removed all

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java (hunk 11 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventPredicates.java]: removed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 24 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 22 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 21 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 20 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 17 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java:36]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java (hunk 18 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventPredicates.java]: removed nullcheck

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java (hunk 15 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java]: fixed

**Comment by stbischof** [comment]
@bjhargrave could you please review again?
